src/components/Article/Detail/ArticleDetailComments.tsx

```diff
@@ -1,49 +1,146 @@
-            ))}
-            {!!remaining && !showMore && (
-              <Divider
-                label={
-                  <Group spacing="xs" align="center">
-                    <Text variant="link" sx={{ cursor: 'pointer' }} onClick={toggleShowMore}>
-                      Show {remaining} More
-                    </Text>
-                  </Group>
-                }
-                labelPosition="center"
-                variant="dashed"
-              />
-            )}
-            {created.map((comment) => (
-              <Comment key={comment.id} comment={comment} />
-            ))}
-          </Stack>
-        )
-      }
-    </CommentsProvider>
+    <>
+      <CommentsProvider
+        entityType="article"
+        entityId={articleId}
+        limit={20}
+        badges={[{ userId, label: 'op', color: 'violet' }]}
+      >
+        {({ data, created, isLoading, remaining, showMore, hiddenCount, toggleShowMore }) =>
+          isLoading ? (
+            <Center mt="xl">
+              <Loader variant="bars" />
+            </Center>
+          ) : (
+            <Stack mt="xl" spacing="xl">
+              <Group spacing="md">
+                <Title order={2} id="comments">
+                  Comments
+                </Title>
+                {hiddenCount > 0 && (
+                  <Button variant="subtle" size="xs" onClick={() => setOpened(true)} compact>
+                    <Group spacing={4} position="center">
+                      <IconMessageCancel size={16} />
+                      <Text inherit inline>
+                        {`See ${hiddenCount} more hidden ${
+                          hiddenCount > 1 ? 'comments' : 'comment'
+                        }`}
+                      </Text>
+                    </Group>
+                  </Button>
+                )}
+              </Group>
+              <CreateComment />
+              {data?.map((comment) => (
+                <Comment key={comment.id} comment={comment} resourceOwnerId={userId} />
+              ))}
+              {!!remaining && !showMore && (
+                <Divider
+                  label={
+                    <Group spacing="xs" align="center">
+                      <Text variant="link" sx={{ cursor: 'pointer' }} onClick={toggleShowMore}>
+                        Show {remaining} More
+                      </Text>
+                    </Group>
+                  }
+                  labelPosition="center"
+                  variant="dashed"
+                />
+              )}
+              {created.map((comment) => (
+                <Comment key={comment.id} comment={comment} resourceOwnerId={userId} />
+              ))}
+            </Stack>
+          )
+        }
+      </CommentsProvider>
+      <HiddenCommentsModal
+        opened={opened}
+        onClose={() => setOpened(false)}
+        entityId={articleId}
+        userId={userId}
+      />
+    </>
   );
 }
+
+type HiddenCommentsModalProps = {
```

Comment:
Sounds like this should live within the comments v2 provider instead of the articles

Reply:
What I mean by this is that this looks reusable enough that should apply to all entities btw

Reply:
Yes, totally. Planning on doing that next